lessons/module-2/closures.md

```diff
@@ -148,15 +157,120 @@ Our most thorough definition of a closure is now **when an inner function has ac
+
+The returned function is a closure, meaning it has access to the taxRate parameter from the outer function even after the outer function has returned. This allows us to "dry up" our code by avoiding duplication of tax calculation logic.
+
+In the subsequent lines of code, we call `calculateTaxes()` twice, once with a taxRate of `0.24` to create a function `taxRateFor2020`, and once with a taxRate of `0.27` to create a function `taxRateFor2021`. We then call each of these functions with different income values to calculate the tax owed.
+
+Using closures in this way allows us to create reusable functions for calculating taxes at different rates, without having to repeat the same logic for each rate. It also allows us to easily calculate taxes for different incomes without having to pass the `taxRate` parameter every time we call the function.
+
 <section class="call-to-action">
 ### Practice
 
-Use the closure pattern to create a "private" counter. The counter variable should be protected by an outter function. The outter function should return an object of methods that allow a user to `incrementCounter`, `decrementCounter` and `getCounterValue`. 
+Consider the following code:
+```js
+function counter() {
+  let count = 0;
+
+  function increment() {
+    count++;
+    console.log(count);
+  }
+
+  increment();
+  increment();
+  increment();
+}
+
+counter(); // Output: 1 2 3
+```
+
+We're using a closure here, which is great! However, when we run our counter function, it only counts to three everytime. Update this function to return a function that allows us to create many counters that keep track of their count separately. 
+
+<section class="answer">
+### Potential Solution
+
+```js
+function createCounter() {
+  let count = 0;
+
+  function increment() {
+    count++;
+    console.log(count);
+  }
+
+  return increment;
+}
+
+const counter = createCounter();
+
+counter(); // Output: 1
+counter(); // Output: 2
+counter(); // Output: 3
+
+const counter2 = createCounter();
+
+counter2(); // Output: 1
+```
+</section>
+***Extra Spicy:*** Include the private methods `incrementCounter`, `decrementCounter` and `getCounterValue` within the closure.
+
 </section>
 
+## Conclusion
+
+ Closures are functions that capture and remember the environment in which they were created, and can be used to create private variables and **memoize** functions. Implementation of closures can result in efficient, modular, and maintainable code, and avoid common pitfalls such as global namespace pollution, code duplication, and side effects. 
+
 <section class="checks-for-understanding">
 ### Checks for Understanding 
 
 - What is lexical scope?
 - What is a closure and how can it be helpful?
+- What is an example of a scenario where closures could be useful in code?
+- What is the difference between a closure and a higher-order function?
```

Comment:
Can you think of an example in your current or recent project where a closure would have been useful?

Reply:
These questions are super important - Being sure that these questions get covered whole group would be great (rather than just - reflect on these questions in your notebook and then ending class)